src/Ryujinx/Modules/Updater/UpdateDialog.cs

```diff
@@ -47,9 +48,14 @@ private void YesButton_Clicked(object sender, EventArgs args)
             if (_restartQuery)
             {
                 string ryuName = OperatingSystem.IsWindows() ? "Ryujinx.exe" : "Ryujinx";
-                string ryuExe  = System.IO.Path.Combine(AppDomain.CurrentDomain.BaseDirectory, ryuName);
 
-                Process.Start(ryuExe, CommandLineState.Arguments);
+                ProcessStartInfo processStart = new(ryuName, string.Join(' ', CommandLineState.Arguments))
+                {
+                    UseShellExecute = true,
+                    WorkingDirectory = ReleaseInformation.GetBaseApplicationDirectory()
+                };
+
+                Process.Start(processStart);
```

Comment:
```suggestion
                ProcessStartInfo processStart = new(ryuName)
                {
                    UseShellExecute = true,
                    WorkingDirectory = ReleaseInformation.GetBaseApplicationDirectory()
                };

                foreach (string argument in CommandLineState.Arguments)
                {
                    processStart.ArgumentList.Add(argument);
                }

                Process.Start(processStart);
```

I think doing something like that would be safer.

Reply:
Done.